tasks/next_date.go

```diff
@@ -0,0 +1,98 @@
+package tasks
+
+import (
+	"errors"
+	"fmt"
+	"go_final_project/str"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func normalizeToDate(t time.Time) time.Time {
+	year, month, day := t.Date()
+	return time.Date(year, month, day, 0, 0, 0, 0, time.UTC)
+}
+
+func isLeapYear(year int) bool {
+	return (year%4 == 0 && year%100 != 0) || (year%400 == 0)
+}
+
+func NextDate(now time.Time, date string, repeat string) (string, error) {
+	if repeat == "" {
+		return "", fmt.Errorf("не указан repeat")
+	}
+
+	nextDate, err := time.ParseInLocation(str.DateFormat, date, time.UTC)
+	if err != nil {
+		return "", fmt.Errorf("неверный формат даты: %v", err)
+	}
+
+	now = normalizeToDate(now.UTC())
+	nextDate = normalizeToDate(nextDate.UTC())
+
+	repeatRule := strings.Fields(repeat)
+
+	switch repeatRule[0] {
+	case "d":
+		if len(repeatRule) < 2 {
+			return "", errors.New("не указано количество дней")
+		}
+
+		days, err := strconv.Atoi(repeatRule[1])
+		if err != nil {
+			return "", fmt.Errorf("неверное значение для дней: %v", err)

```

Comment:
Когда просто выводим ошибку в лог, то `%v`, а тут мы ее оборачиваем (wrap)
```suggestion
			return "", fmt.Errorf("неверное значение для дней: %w", err)
```